resources/js/features/achievements/components/UnlockableAchievementAvatar/UnlockableAchievementAvatar.test.tsx

```diff
@@ -0,0 +1,135 @@
+import { render, screen } from '@/test';
+import { createAchievement, createGame } from '@/test/factories';
+
+import { UnlockableAchievementAvatar } from './UnlockableAchievementAvatar';
+
+describe('Component: UnlockableAchievementAvatar', () => {
+  it('renders without crashing', () => {
+    // ARRANGE
+    const achievement = createAchievement();
+    const { container } = render(<UnlockableAchievementAvatar achievement={achievement} />);
+
+    // ASSERT
+    expect(container).toBeTruthy();
+  });
+
+  it('displays a locked achievement correctly', () => {
+    // ARRANGE
+    const achievement = createAchievement({
+      title: 'Creative Name',
+      description: 'Do the thing',
+    });
+
+    render(<UnlockableAchievementAvatar achievement={achievement} />);
+
+    // ASSERT
+    const img = screen.getByRole('img') as HTMLImageElement;
```

Comment:
Great work on these test suites. One way to get around the `as HTMLImageElement` type cast is to use `toHaveAttribute()`:
```ts
expect(img).toHaveAttribute('src', achievement.badgeLockedUrl);
```

Reply:
Provided cast was found via google search. Do you want it changed?

Reply:
To be honest, I don't have a strong opinion beyond the philosophical. The current approach as implemented is fine.